jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/sender/JMSClientConnector.java

```diff
@@ -0,0 +1,175 @@
+
+/**
+ * JMS sender implementation.
+ */
+
+public class JMSClientConnector implements ClientConnector {
+
+    private MessageProducer messageProducer;
+    private Session session;
+    private Connection connection;
+    private JMSConnectionFactory jmsConnectionFactory;
+
+    @Override public boolean send(CarbonMessage carbonMessage, CarbonCallback carbonCallback)
+            throws ClientConnectorException {
+        return false;
+    }
+
+    @Override public boolean send(CarbonMessage carbonMessage, CarbonCallback carbonCallback,
+                                  Map<String, String> propertyMap) throws ClientConnectorException {
+        try {
+            try {
+                Set<Map.Entry<String, String>> propertySet = propertyMap.entrySet();
+                this.createConnection(propertySet);
+            } catch (JMSConnectorException e) {
+                throw new ClientConnectorException(e.getMessage(), e);
+            }
+
+            Message message = null;
+            String messageType = propertyMap.get(JMSConstants.JMS_MESSAGE_TYPE);
+
+            if (carbonMessage instanceof TextCarbonMessage) {
+                String textData = ((TextCarbonMessage) carbonMessage).getText();
+                if (messageType.equals(JMSConstants.TEXT_MESSAGE_TYPE)) {
+                    message = session.createTextMessage();
+                    TextMessage textMessage = (TextMessage) message;
+                    textMessage.setText(textData);
+                } else if (messageType.equals(JMSConstants.BYTES_MESSAGE_TYPE)) {
+                    message = session.createBytesMessage();
+                    BytesMessage bytesMessage = (BytesMessage) message;
+                    bytesMessage.writeBytes(textData.getBytes(Charset.defaultCharset()));
+                }
+            } else if (messageType.equals(JMSConstants.OBJECT_MESSAGE_TYPE) &&
+                       carbonMessage instanceof SerializableCarbonMessage) {
+                message = session.createObjectMessage((SerializableCarbonMessage) carbonMessage);
+            } else if (messageType.equals(JMSConstants.MAP_MESSAGE_TYPE) && carbonMessage instanceof MapCarbonMessage) {
+                message = session.createMapMessage();
+                MapMessage mapMessage = (MapMessage) message;
+                Enumeration<String> mapNames = ((MapCarbonMessage) carbonMessage).getMapNames();
+                while (mapNames.hasMoreElements()) {
+                    String key = mapNames.nextElement();
+                    mapMessage.setString(key, ((MapCarbonMessage) carbonMessage).getValue(key));
+                }
+            }
+
+            if (carbonMessage.getProperty(JMSConstants.PERSISTENCE) != null &&
+                carbonMessage.getProperty(JMSConstants.PERSISTENCE).equals(false)) {
+                messageProducer.setDeliveryMode(DeliveryMode.NON_PERSISTENT);
+            }
+
+            if (message != null) {
+                messageProducer.send(message);
```

Comment:
It becomes null if message type is not defined by the user. Anyway it cannot happen in the current scenario since it is validated at the application level. But i will anyway throw a ClientConnectorException if message is null.